### PR TITLE
Fix arg inputs.

### PR DESCRIPTION
**Pull request by ericsciple**
Is the fix already in master?  If not, link the bug tracking that work.
No. I linked the M99 bug to the bug for master.

What testing was done?
Manual verification no targets, one target, two targets.

Does this fix change the AT/DT contract?
No.

Does this fix include any database servicing changes?
Task upload step to deploy the hotfix.

Were tests added to cover this to make sure it doesn't regress?  If not, link a bug or user story tracking that work.
No. I added a link to the test case issue from the M99 bug.

### CODE REVIEW

File: Tasks/Npm/npmtask.ts [hunk 1]
```diff
@@ -12,6 +12,9 @@ tl.mkdirP(cwd);
 tl.cd(cwd);
 
 var command = tl.getInput('command', true);
+if (command.indexOf(' ') >= 0) {
+	tl.setResult(tl.TaskResult.Failed, tl.loc("InvalidCommand"));
```
Comment (TingluoHuang) [inline Tasks/Npm/npmtask.ts]: tl.loc("InvalidCommand", command)

File: Tasks/Grunt/grunttask.ts [hunk 2]
```diff
@@ -33,7 +33,13 @@ else {
 }
 
 // optional - no tasks will concat nothing
-gt.arg(tl.getInput('targets', false));
+tl.getDelimitedInput('targets', ' ', false)
+	.forEach(x => {
```
Comment (bryanmacfarlane) [inline Tasks/Grunt/grunttask.ts]: let's do this in task lib in 100.  Change should be just getDelimited inputs

File: Tasks/Gulp/gulptask.ts [hunk 3]
```diff
@@ -33,7 +33,13 @@ else {
 }
 
 // optional - no tasks will concat nothing
-gt.arg(tl.getInput('targets', false));
+tl.getDelimitedInput('targets', ' ', false)
+	.forEach(x => {
+		// omit empty values
```
Comment (bryanmacfarlane) [inline Tasks/Gulp/gulptask.ts]: same

File: Tasks/Npm/Strings/resources.resjson/en-US/resources.resjson [hunk 4]
```diff
@@ -10,6 +10,7 @@
   "loc.input.help.arguments": "Additional arguments passed to npm.",
   "loc.input.label.cwd": "Working Directory",
   "loc.input.help.cwd": "Current working directory where npm command is run.  Defaults to the root of the repo.",
+  "loc.messages.InvalidCommand": "Illegal characters in command '%s'. Command cannot contain a space.",
```
Comment (bryanmacfarlane) [inline Tasks/Npm/Strings/resources.resjson/en-US/resources.resjson]: Only one command should be used for npm.  Use arguments input for additional arguments.

File: Tasks/Npm/npmtask.ts (hunk 1 above)
Comment (ericsciple) [inline Tasks/Npm/npmtask.ts]: Fixed. Changed the error message to not require a parameter :)

**Comment by ericsciple** [comment]
@lafritay @bryanmacfarlane

**Comment by bryanmacfarlane** [comment]
Looks good

**Pull request merged**